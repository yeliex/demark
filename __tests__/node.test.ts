import { Processor } from '../src/index.js';

describe('Output as Node', () => {
    const processor = new Processor();

    test('should output node', async () => {
        const result = processor.toNode(
            `# Hello World

## sub

- 1
- 2

`,
        );

        expect(result).toMatchSnapshot();
    });

    test('should support HTML tag', async () => {
        const result = processor.toNode(
            `# Hello World

<test string="string" number={1} array={[1, 'string', {obj: true}]} object={{obj: true}} />
`,
        );

        expect(result).toMatchSnapshot();
    });

    test('should support HTML tag with children', async () => {
        const result = processor.toNode(
            `# Hello World

<test>
children content
</test>
`,
        );

        expect(result).toMatchSnapshot();
    });

    test('should support HTML tag with children markdown', async () => {
        const result = processor.toNode(
            `# Hello World

<test>
> blockquote
</test>
`,
        );

        expect(result).toMatchSnapshot();
    });
});
