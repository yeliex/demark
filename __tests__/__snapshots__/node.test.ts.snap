// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Output as Node should output node 1`] = `
{
  "data": [
    {
      "children": [
        {
          "type": "text",
          "value": "Hello World",
        },
      ],
      "properties": {
        "id": "hello-world",
      },
      "tagName": "h1",
      "type": "element",
    },
    {
      "children": [
        {
          "type": "text",
          "value": "sub",
        },
      ],
      "properties": {
        "id": "sub",
      },
      "tagName": "h2",
      "type": "element",
    },
    {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "1",
            },
          ],
          "properties": {},
          "tagName": "li",
          "type": "element",
        },
        {
          "children": [
            {
              "type": "text",
              "value": "2",
            },
          ],
          "properties": {},
          "tagName": "li",
          "type": "element",
        },
      ],
      "properties": {},
      "tagName": "ul",
      "type": "element",
    },
  ],
  "title": "Hello World",
  "toc": undefined,
}
`;

exports[`Output as Node should support HTML tag 1`] = `
{
  "data": [
    {
      "children": [
        {
          "type": "text",
          "value": "Hello World",
        },
      ],
      "properties": {
        "id": "hello-world",
      },
      "tagName": "h1",
      "type": "element",
    },
    {
      "children": [],
      "name": "test",
      "properties": {
        "array": [
          1,
          "string",
          {
            "obj": true,
          },
        ],
        "number": 1,
        "object": {
          "obj": true,
        },
        "string": "string",
      },
      "type": "component",
    },
  ],
  "title": "Hello World",
  "toc": undefined,
}
`;

exports[`Output as Node should support HTML tag with children 1`] = `
{
  "data": [
    {
      "children": [
        {
          "type": "text",
          "value": "Hello World",
        },
      ],
      "properties": {
        "id": "hello-world",
      },
      "tagName": "h1",
      "type": "element",
    },
    {
      "children": [
        {
          "children": [
            {
              "type": "text",
              "value": "children content",
            },
          ],
          "properties": {},
          "tagName": "p",
          "type": "element",
        },
      ],
      "name": "test",
      "properties": {},
      "type": "component",
    },
  ],
  "title": "Hello World",
  "toc": undefined,
}
`;

exports[`Output as Node should support HTML tag with children markdown 1`] = `
{
  "data": [
    {
      "children": [
        {
          "type": "text",
          "value": "Hello World",
        },
      ],
      "properties": {
        "id": "hello-world",
      },
      "tagName": "h1",
      "type": "element",
    },
    {
      "children": [
        {
          "children": [
            {
              "children": [
                {
                  "type": "text",
                  "value": "blockquote",
                },
              ],
              "properties": {},
              "tagName": "p",
              "type": "element",
            },
          ],
          "properties": {},
          "tagName": "blockquote",
          "type": "element",
        },
      ],
      "name": "test",
      "properties": {},
      "type": "component",
    },
  ],
  "title": "Hello World",
  "toc": undefined,
}
`;
